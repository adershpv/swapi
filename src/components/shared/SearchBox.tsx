import { Input } from "@nextui-org/react";
import { SearchIcon } from "../icons/SearchIcon";

interface SearchBoxProps {
  value: string;
  onSearchChange: (value: string) => void;
}

export const SearchBox: React.FC<SearchBoxProps> = ({
  value,
  onSearchChange,
}) => {
  return (
    <Input
      value={value}
      onChange={(e) => onSearchChange(e.target.value)}
      autoComplete="off"
      classNames={{
        inputWrapper: [
          "bg-white",
          "data-[hover=true]:bg-white",
          "group-data-[focus=true]:bg-white",
        ],
      }}
      label="Search"
      placeholder="Type to search..."
      radius="lg"
      startContent={
        <SearchIcon className="text-black/50 mb-0.5 dark:text-white/90 text-slate-400 pointer-events-none flex-shrink-0" />
      }
    />
  );
};
