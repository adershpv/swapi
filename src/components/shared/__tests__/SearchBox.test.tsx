import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { SearchBox } from "../SearchBox";

describe("SearchBox Component", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(
      <SearchBox value="test" onSearchChange={jest.fn()} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("calls onSearchChange when the input value changes", () => {
    const mockOnSearchChange = jest.fn();
    render(<SearchBox value="" onSearchChange={mockOnSearchChange} />);

    const input = screen.getByLabelText("Search");
    fireEvent.change(input, { target: { value: "new value" } });

    expect(mockOnSearchChange).toHaveBeenCalledTimes(1);
    expect(mockOnSearchChange).toHaveBeenCalledWith("new value");
  });
});
