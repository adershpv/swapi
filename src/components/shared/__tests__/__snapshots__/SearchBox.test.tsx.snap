// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="group flex flex-col data-[hidden=true]:hidden w-full is-filled"
    data-filled="true"
    data-filled-within="true"
    data-has-elements="true"
    data-has-label="true"
    data-has-value="true"
    data-slot="base"
  >
    <div
      class="relative w-full inline-flex tap-highlight-transparent shadow-sm px-3 min-h-10 rounded-large flex-col items-start justify-center gap-0 transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background h-14 py-2 bg-white data-[hover=true]:bg-white group-data-[focus=true]:bg-white is-filled"
      data-slot="input-wrapper"
      style="cursor: text;"
    >
      <label
        class="absolute z-10 pointer-events-none origin-top-left flex-shrink-0 rtl:origin-top-right subpixel-antialiased block text-foreground-500 cursor-text will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] group-data-[filled-within=true]:text-default-600 group-data-[filled-within=true]:pointer-events-auto group-data-[filled-within=true]:scale-85 text-small group-data-[filled-within=true]:-translate-y-[calc(50%_+_theme(fontSize.small)/2_-_6px)] pe-2 max-w-full text-ellipsis overflow-hidden"
        data-slot="label"
        for="react-aria-:r0:"
        id="react-aria-:r1:"
      >
        Search
      </label>
      <div
        class="inline-flex w-full items-center h-full box-border group-data-[has-label=true]:items-end pb-0.5"
        data-slot="inner-wrapper"
      >
        <svg
          aria-hidden="true"
          class="text-black/50 mb-0.5 dark:text-white/90 text-slate-400 pointer-events-none flex-shrink-0"
          fill="none"
          focusable="false"
          height="1em"
          role="presentation"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M11.5 21C16.7467 21 21 16.7467 21 11.5C21 6.25329 16.7467 2 11.5 2C6.25329 2 2 6.25329 2 11.5C2 16.7467 6.25329 21 11.5 21Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M22 22L20 20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        <input
          aria-label="Search"
          aria-labelledby="react-aria-:r0: react-aria-:r1:"
          autocomplete="off"
          class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-small group-data-[has-value=true]:text-default-foreground is-filled"
          data-filled="true"
          data-filled-within="true"
          data-has-start-content="true"
          data-slot="input"
          id="react-aria-:r0:"
          placeholder="Type to search..."
          title=""
          type="text"
          value="test"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
